src/sentry/api/endpoints/organization_release_health_data.py

```diff
@@ -63,6 +63,10 @@ def data_fn(offset: int, limit: int):
                 # due to possible data corruption crash_free_rate value can be less than 0,
                 # which is not valid behavior, so those values have to be bottom capped at 0
                 metrics.ensure_non_negative_crash_free_rate_value(data, request, organization)
+                # due to possible data corruption crash_rate value can be greater than 1,
+                # which is not valid behavior, so those values have to be capped at 1
+                metrics.ensure_crash_rate_not_greater_than_1(data, request, organization)
```

Comment:
can we do something like `metrics.ensure_crash_rate_in_bounds` that performs both checks so that we do not iterate twice over the response?

Reply:
There is no much overhead in iterating through the response few times (usually number of groups in that API endpoint is 1 or 2, rarely over that), and every other iteration is key specific, which means it won't happen unless a metric key for which we do checks is present.

I could combine all of this condition into one function which checks everything, but it would make the code much more complex and I would prefer approach with simpler and cleaner code, but slight overhead of few iterations more.

I have refactored this code to use only one function for all crash rate bound checks, so it is easier to maintain and refactor later.

Reply:
Yeah, agreed, this is great. Thanks!